x/leveragelp/keeper/msg_server_remove_pool.go

```diff
@@ -0,0 +1,26 @@
+package keeper
+
+import (
+	"context"
+
+	errorsmod "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	"github.com/elys-network/elys/x/leveragelp/types"
+)
+
+func (k msgServer) RemovePool(goCtx context.Context, msg *types.MsgAddPool) (*types.MsgAddPoolResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if k.authority != msg.Authority {
+		return nil, errorsmod.Wrapf(govtypes.ErrInvalidSigner, "invalid authority; expected %s, got %s", k.authority, msg.Authority)
+	}
+
+	_, found := k.GetPool(ctx, msg.Pool.AmmPoolId)
+
```

Comment:
Please add some checks here, we should not be able to delete Pool with lpAmount > 0